### PR TITLE
Update mlx5 driver to v4.6

### PR DESCRIPTION
**Pull request by andrewbonney**
This PR updates the existing patches for the Mellanox Ethernet driver to v4.6.

### CODE REVIEW

**Comment by giuseppelettieri** [comment]
Thanks!

**Comment by kierank** [comment]
Do you plan to have the Makefile pull the 4.6 driver?

File: LINUX/final-patches/mellanox--mlx5--4.6 [hunk 1]
```diff
@@ -0,0 +1,350 @@
...
++
+ static inline bool mlx5e_rx_hw_stamp(struct hwtstamp_config *config)
+ {
+ 	return config->rx_filter == HWTSTAMP_FILTER_ALL;
+@@ -154,7 +162,7 @@ static inline u32 mlx5e_decompress_cqes_cont(struct mlx5e_rq *rq,
+ 	return cqe_count;
+ }
+ 
+-static inline u32 mlx5e_decompress_cqes_start(struct mlx5e_rq *rq,
++u32 mlx5e_decompress_cqes_start(struct mlx5e_rq *rq,
```
Comment (kierank) [inline LINUX/final-patches/mellanox--mlx5--4.6]: @andrewbonney This is a stray change, right? Doesn't seem to do anything

Comment (andrewbonney) [reply LINUX/final-patches/mellanox--mlx5--4.6]: Quite possibly. This appeared in the 4.5 patch too so I suspect it's just been brought forward into this one without noticing.

Comment (kierank) [reply LINUX/final-patches/mellanox--mlx5--4.6]: Ah, it's causing problems when patching 4.7 so I'll drop it in that patchset

Comment (kierank) [reply LINUX/final-patches/mellanox--mlx5--4.6:276]: Does seem it's needed otherwise you get an error about a missing mlx5e_decompress_cqes_start symbol

**Pull request merged**